homeassistant/components/fan/alfawise.py

```diff
@@ -0,0 +1,95 @@
+"""
+Alfawise fan platform that has a fake fan.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/fan.alfawise/
+"""
+import logging
+
+from homeassistant.components.fan import (SPEED_LOW, SPEED_HIGH,
+                                          FanEntity, SUPPORT_SET_SPEED)
+from homeassistant.const import CONF_DEVICES, CONF_NAME, CONF_MAC
+from homeassistant.const import STATE_OFF
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['pyAlfawise==0.4-beta']
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Alfawise fan platform."""
+    fans = []
+
+    for ipaddr, device_config in config[CONF_DEVICES].items():
+        name = device_config[CONF_NAME]
+        macaddr = device_config[CONF_MAC]
+        _LOGGER.debug("Adding configured %s", name)
+
+        fan = AlfawiseFan(hass, ipaddr, macaddr, name)
+        fans.append(fan)
+
+    add_devices(fans, True)
+
+
+class AlfawiseFan(FanEntity):
+    """A Alfawisenstration fan component."""
+
+    def __init__(self, hass, ipaddr, macaddr, name):
+        """Initialize the entity."""
+        import pyAlfawise
+        self.hass = hass
+        self._speed = STATE_OFF
+        self._name = name
+        self._fan = pyAlfawise.Alfawise(macaddr, ipaddr)
```

Comment:
Move this to `setup_platform()` and pass it to `AlfawiseFan()`. Check during the platform setup if the device is present/reachable and decide what to do (`PlatformNotReady` or make the setup fail).